src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Order/TreeItemOrderPropertyProvider.cs

```diff
@@ -48,13 +58,14 @@ private void ComputeIndices()
             // _displayOrderMap only.
             foreach (var item in OrderedItems)
             {
-                var includeParts = item.EvaluatedInclude.Split(CommonConstants.PathSeparators, StringSplitOptions.RemoveEmptyEntries);
+                var includeParts = GetStringParts(item.EvaluatedInclude);
                 foreach (var part in includeParts)
                 {
-                    var rootedPath = duplicateFiles.Contains(part) ? _project.MakeRooted(item.EvaluatedInclude) : null;
-                    if (rootedPath != null && !_rootedOrderMap.ContainsKey(rootedPath))
+                    // EvaluatedInclude is a unique relative path for a file.
```

Comment:
EvaluatedInclude does not have to be relative, it can be rooted; "C:\Foo.cs" or current directory relative "C:Foo.cs".

Reply:
In the second PR for CPS, I ensure that I only give the relative path. This makes sense as I only care about how a file or folder gets rendered relative to the project; that's the path I actually want and the ones we care about ordering. Perhaps, in CPS, ```ProjectItemIdentity``` property "EvaluatedInclude" is a bad name. If it were really "EvaluatedInclude", I would never get the proper link path to do ordering.

Reply:
To note, something like this:
```
<Compile Include="../../ConsoleApp2.sln" />
```
Will actually turn into a linked file, with a link path of just "ConsoleApp2.sln".

Reply:
This won't turn in a linked file:
```
<Compile Include="C:\Projects\Project\Program.cs" />
```

When the csproj lives in C:\Projects\Project.

Reply:
It will not, but in my CPS PR, it will return "Program.cs".

Reply:
You can't change the identity of an item.

Reply:
I think we need to sync up with CPS/offline. You changing some fundamental details of how MSBuild works here. EvaluatedInclude is the identity of an item - you can't make that something else.